pkg/proxy/iptables/proxier.go

```diff
@@ -1126,24 +1126,22 @@ func (proxier *Proxier) syncProxyRules() {
 				)
 
 				destChain := svcXlbChain
-				// We have to SNAT packets to external IPs if externalTrafficPolicy is cluster.
+				// We have to SNAT packets to external IPs if externalTrafficPolicy is cluster
+				// and the traffic is NOT Local. Local traffic coming from Pods and Nodes will
+				// be always sent directly to the corresponding Service, so no need to SNAT
 				if !svcInfo.OnlyNodeLocalEndpoints() {
 					destChain = svcChain
-					writeLine(proxier.natRules, append(args, "-j", string(KubeMarkMasqChain))...)
+					if proxier.localDetector.IsImplemented() {
```

Comment:
Shouldn't you also jump to MASQ if it is not implemented?  Pessimistic

Reply:
yep